src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -114,6 +116,18 @@ internal static FunctionMetadata ReadFunctionMetadata(string scriptDir, ICollect
                 }
                 else if (functionMetadata != null)
                 {
+                    if (functionMetadata.ScriptType == ScriptType.Unknown)
```

Comment:
Can we move this logic to the `ParseScriptType` method so that it's all in the same place?